src/MagnumPlugins/BasisImporter/BasisImporter.cpp

```diff
@@ -0,0 +1,217 @@
+                : CompressedPixelFormat::Bc1RGBUnorm;
+        case BasisTranscodingType::Bc3:
+            CORRADE_ASSERT(hasAlpha, "Bc3 always has alpha, but non-alpha was requested",
+                CompressedPixelFormat::Bc3RGBAUnorm);
+            return CompressedPixelFormat::Bc3RGBAUnorm;
+        case BasisTranscodingType::Bc4:
+            return CompressedPixelFormat::Bc4RUnorm;
+        case BasisTranscodingType::Bc5:
+            return CompressedPixelFormat::Bc5RGUnorm;
+        case BasisTranscodingType::Bc7M6OpaqueOnly:
+            CORRADE_ASSERT(hasAlpha, "Bc7 is opaque only, but alpha was requested",
+                CompressedPixelFormat::Bc7RGBAUnorm);
+            return CompressedPixelFormat::Bc7RGBAUnorm;
+        case BasisTranscodingType::Pvrtc1_4OpaqueOnly:
+            CORRADE_ASSERT(hasAlpha, "Pvrtc1 is opaque only, but alpha was requested",
+                CompressedPixelFormat::PvrtcRGB4bppUnorm);
+            return CompressedPixelFormat::PvrtcRGB4bppUnorm;
+        default:
+            CORRADE_ASSERT_UNREACHABLE();
+    }
+}
+
+basist::transcoder_texture_format getTranscoderTextureFormat(BasisTranscodingType type) {
+    constexpr basist::transcoder_texture_format formats[] = {
+        basist::transcoder_texture_format::cTFETC1,
+        basist::transcoder_texture_format::cTFETC2,
+        basist::transcoder_texture_format::cTFBC1,
+        basist::transcoder_texture_format::cTFBC3,
+        basist::transcoder_texture_format::cTFBC4,
+        basist::transcoder_texture_format::cTFBC5,
+        basist::transcoder_texture_format::cTFBC7_M6_OPAQUE_ONLY,
+        basist::transcoder_texture_format::cTFPVRTC1_4_OPAQUE_ONLY
+    };
+
+    return formats[Int(type)];
+}
+
+}
+
+struct BasisImporter::State {
+    /* There is only this type of codebook */
+    basist::etc1_global_selector_codebook _codebook;
+    basist::basisu_transcoder _transcoder{&_codebook};
+
+    State(): _codebook(basist::g_global_selector_cb_size,
+        basist::g_global_selector_cb) {}
+};
+
+BasisImporter::BasisImporter() = default;
+
+BasisImporter::BasisImporter(PluginManager::AbstractManager& manager, const std::string& plugin): AbstractImporter{manager, plugin} {
+    /* Does nothing if already initialized */
+    basist::basisu_transcoder_init();
+
+    /* Initializes codebook */
+    _state.reset(new State);
+
+    /* Set format configuration from plugin alias
+       or set a default */
+    if(Corrade::Utility::String::beginsWith(plugin, "BasisImporter")) {
+
+        /* Has type prefix */
+        if(plugin.length() > 13) {
+            /* We can assume the substring results in a valid value
+               as the plugin conf limits it to known suffixes */
+            configuration().setValue("format", plugin.substr(13));
+
+        /* Default configuration to Etc2 */
+        } else {
+            configuration().setValue("format",
+                BasisTranscodingType::Etc2);
```

Comment:
This default ~~could~~ should be done directly inside the `BasisImporter.conf` so people can modify it by hand, when needed:

```ini
provides=BasisImporterEtc1
provides=BasisImporterEtc2
provides=BasisImporterBc1
provides=BasisImporterBc3
provides=BasisImporterBc4
provides=BasisImporterBc5
provides=BasisImporterBc7M6OpaqueOnly
provides=BasisImporterPvrtc1_4OpaqueOnly

format=Etc2
```